Mention gccemacs in lsp-doctor and in performance guide (#2213)
emacs-lsp/lsp-mode: docs/page/performance.md

``` elisp
(setq lsp-idle-delay 0.500)
```
### gccemacs

> [gccemacs](https://akrl.sdf.org/gccemacs.html) is the bleading edge version of `Emacs` which byte-compiles elisp to native code resulting in more than 200% speedup. For everyone looking for optimal performance `gccemacs` is the way to go.

Reviewer: ```suggestion
[gccemacs](https://akrl.sdf.org/gccemacs.html) is a bleeding-edge version of `Emacs` that compiles elisp to native code, resulting in more than 200% speedup. For everyone looking for optimal performance, `gccemacs` is the way to go.
```
